app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/MixedUseController.scala

```diff
@@ -0,0 +1,429 @@
+                             )
+                           )
+                         )
+                       )
+                } yield ()
+
+                result.fold(
+                  e => errorHandler.errorResult(),
+                  _ =>
+                    Redirect(
+                      routes.MixedUseController.checkYourAnswers()
+                    )
+                )
+              }
+          )
+      }
+    }
+
+  def enterAcquisitionValue(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val answers = r.draftReturn.mixedUsePropertyDetailsAnswers
+          .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
+
+        val backLink = routes.MixedUseController.enterDisposalValue()
+
+        val acquisitionPrice = answers
+          .fold(_.acquisitionPrice, c => Some(c.acquisitionPrice))
+
+        val form = acquisitionPrice.fold(acquisitionPriceForm)(c => acquisitionPriceForm.fill(c.inPounds))
+
+        Ok(
+          acquisitionValuePage(
+            form,
+            backLink,
+            r.journey.subscribedDetails.isATrust,
+            r.representativeType
+          )
+        )
+      }
+    }
+
+  def enterAcquisitionValueSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val answers  = r.draftReturn.mixedUsePropertyDetailsAnswers
+          .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
+        val backLink = routes.MixedUseController.enterDisposalValue()
+
+        acquisitionPriceForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              BadRequest(
+                acquisitionValuePage(
+                  formWithErrors,
+                  backLink,
+                  r.journey.subscribedDetails.isATrust,
+                  r.representativeType
+                )
+              ),
+            acquisitionPrice =>
+              if (
+                answers
+                  .fold(_.acquisitionPrice, c => Some(c.acquisitionPrice))
+                  .contains(AmountInPence.fromPounds(acquisitionPrice))
+              )
+                Redirect(
+                  routes.MixedUseController.checkYourAnswers()
+                )
+              else {
+                val updatedAnswers     =
+                  answers
+                    .fold(
+                      _.copy(acquisitionPrice = Some(AmountInPence.fromPounds(acquisitionPrice))),
+                      _.copy(acquisitionPrice = AmountInPence.fromPounds(acquisitionPrice))
+                    )
+                val updatedDraftReturn = r.draftReturn.copy(
+                  mixedUsePropertyDetailsAnswers = Some(updatedAnswers),
+                  yearToDateLiabilityAnswers = None
+                )
+                val result             = for {
```

Comment:
you could consider factoring out this `for` comprehension out into a method - it'll save a lot of space as it's being used in multiple places. You could call it something like `updateDraftReturnAndSession` or something